crates/router/src/routes/routing.rs

```diff
@@ -204,6 +214,50 @@ pub async fn list_routing_configs(
     .await
 }
 
+#[cfg(feature = "olap")]
+#[instrument(skip_all)]
+pub async fn list_routing_configs_for_profile(
+    state: web::Data<AppState>,
+    req: HttpRequest,
+    query: web::Query<RoutingRetrieveQuery>,
+    path: web::Path<common_utils::id_type::ProfileId>,
+    transaction_type: &enums::TransactionType,
+) -> impl Responder {
+    let flow = Flow::RoutingRetrieveDictionary;
+    let path = path.into_inner();
+    Box::pin(oss_api::server_wrap(
+        flow,
+        state,
+        &req,
+        query.into_inner(),
+        |state, auth: auth::AuthenticationData, query_params, _| {
+            routing::retrieve_merchant_routing_dictionary(
+                state,
+                auth.merchant_account,
+                auth.profile_id,
```

Comment:
Should pass a vec.

Reply:
https://github.com/juspay/hyperswitch/blob/379831932a9c1e73b669f40af75cd02e682a0ec7/crates/router/src/routes/payments.rs#L1049